utils/hv-tools/files/etc/init.d/hv_kvp_daemon

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh /etc/rc.common
+
+START=21
+
+USE_PROCD=1
+NAME=hv_kvp_daemon
```

Comment:
dtto

Reply:
The hv_kvp_daemon is a service that is part of the Hyper-V Guest Tools and is responsible for the Key-Value Pair (KVP) exchange functionality between a guest virtual machine (VM) and the Hyper-V host.

What does hv_kvp_daemon do?
Key-Value Pair Exchange:

The hv_kvp_daemon facilitates the exchange of key-value pairs between the Hyper-V host and the guest operating system (in this case, OpenWrt).
It enables the host to retrieve specific information about the guest VM, such as network configuration, and also allows the host to set or modify certain configurations within the guest.
Usage in Virtualized Environments:

This daemon is critical for managing the guest VM from the Hyper-V host. For example, it can be used to obtain the guest's IP address or DNS settings, or to inject configuration parameters from the host into the guest VM.
The KVP functionality is often used for monitoring, automation, and integration tasks in a virtualized environment.
Integration with Hyper-V:

The hv_kvp_daemon works in conjunction with other Hyper-V Integration Services to ensure that the guest VM can communicate and exchange data effectively with the Hyper-V host. This integration helps in maintaining and managing the virtual machines more efficiently.
Why is hv_kvp_daemon needed in OpenWrt?
Including this daemon in OpenWrt allows OpenWrt virtual machines running on Hyper-V to participate in the key-value pair exchange with the Hyper-V host. This capability is essential for tasks such as automated management, dynamic configuration, and retrieving important information about the guest VM from the host.

Reply:
removed NAME=hv_kvp_daemon